images/base-glibc-busybox-bash/Dockerfile

```diff
@@ -45,7 +46,18 @@ RUN rm -rf ./lib ./lib64
 COPY install-pkgs /usr/local/bin
 RUN install-pkgs "$( pwd )" /tmp/work bash ncurses-base libc-bin zlib1g
 
+ARG tini_version=0.19.0
+ADD "https://github.com/krallin/tini/releases/download/v${tini_version}/tini-amd64" ./usr/bin/tini
```

Comment:
tini is integrated in Docker nowadays, isn't it? Do you leave this in for Singularity?

Reply:
Ah, good catch! I forgot (<- lost on a rebase before push -.-) to commit a comment here:
```
TODO: Decide whether to include a small init process (tini) for default signal
      handling/reaping. Considerations:
        - `docker/podman run` can do this themselves via the `--init` flag.
          (IIUC) `singularity run --pid` (without `--no-init`) behaves similar.
        - Those `--init`/`--pid` arguments are not defaults.
          - By including `tini` in the image this is made the default.
          - Are there users that explicitly do not want a proper PID 1?
```

Reply:
Of course, we can remove it if you think forcing such default could be problematic (see `Are there users that explicitly do not want a proper PID 1?` above).

Reply:
(Also, good catch on the zlib! I only used it for the test and simply forgot to remove this.)

Reply:
nucurses can also go, left?

Reply:
I don't want to decide for users or use-cases, but if we design a base image for 50.000 containers we should imho have as less as possible assumptions and since Docker, Singularity and co do support this use case natively, why should we force this on a user?

Reply:
> nucurses can also go, left?

We only include `libtinfo5` (because `bash` links to `libtinfo.so.5`) which in turn brings in `ncurses-base`, which essentially just includes some `terminfo` files (but only a selection, not the multi-megabyte ones you usually get with distributions).

Reply:
> I don't want to decide for users or use-cases, but if we design a base image for 50.000 containers we should imho have as less as possible assumptions and since Docker, Singularity and co do support this use case natively, why should we force this on a user?

Fine by me :)

Reply:
In my old container, I had a statically compiled bash binary, maybe that works here too? And would enable us to include bash5 as well?

Reply:
> In my old container, I had a statically compiled bash binary, maybe that works here too?

The non-static `bash` + `libtinfo5` is actually smaller than the static one ;).

> And would enable us to include bash5 as well?

We could just use `debian:10-slim` instead of `debian:9-slim` then. I switched to the latter to make the small and large base image more similar. (I tried `debian:10-slim` before, actually. It works perfectly fine for the smaller one, but I ran into https://github.com/bioconda/bioconda-extended-base-image/pull/10#issue-300762489 when I added libgl for the larger :/.)
Do you have any specific benefit in mind for Bash 5?

Reply:
> The non-static bash + libtinfo5 is actually smaller than the static one ;).

I see, I'm just worried that the shared lib (ncurses) infers with conda stuff, or that we do not test for ncurses anymore etc ...

I'm fine with bash4, that was just an idea, but thanks for reminding me about the bigger Debian 10 image.

Reply:
> I see, I'm just worried that the shared lib (ncurses) infers with conda stuff, or that we do not test for ncurses anymore etc ...

I'd assume it wouldn't be too bad since it's just terminfo and not the whole ncurses.(?)
If you do have concerns, we could also simply move the `libtinfo*.so*` files to another path and add an RPATH to `bash` so it will find it, but nothing else would. WDYT?

> I'm fine with bash4, that was just an idea, but thanks for reminding me about the bigger Debian 10 image.

I just tried this out:
```
~ >>>  podman run --rm -it bitnami/minideb:jessie
root@28bb028f8fc6:/# du --human-readable --summarize --exclude=/proc /
62M	/
root@28bb028f8fc6:/# apt-get update -qq 2>/dev/null
root@28bb028f8fc6:/# apt-get install --yes --no-install-recommends libgl1-mesa-glx >/dev/null 2>&1
root@28bb028f8fc6:/# rm -rf /var/lib/apt/lists/*
root@28bb028f8fc6:/# du --human-readable --summarize --exclude=/proc /
68M	/
root@28bb028f8fc6:/# exit

~ >>>  podman run --rm -it bitnami/minideb:stretch
root@a23483dfe6e4:/# du --human-readable --summarize --exclude=/proc /
63M	/
root@a23483dfe6e4:/# apt-get update -qq 2>/dev/null
root@a23483dfe6e4:/# apt-get install --yes --no-install-recommends libgl1-mesa-glx >/dev/null 2>&1
root@a23483dfe6e4:/# rm -rf /var/lib/apt/lists/*
root@a23483dfe6e4:/# du --human-readable --summarize --exclude=/proc /
70M	/
root@a23483dfe6e4:/# exit

~ >>>  podman run --rm -it debian:9-slim
root@c228adecd486:/# du --human-readable --summarize --exclude=/proc /
65M	/
root@c228adecd486:/# apt-get update -qq 2>/dev/null
root@c228adecd486:/# apt-get install --yes --no-install-recommends libgl1-mesa-glx >/dev/null 2>&1
root@c228adecd486:/# rm -rf /var/lib/apt/lists/*
root@c228adecd486:/# du --human-readable --summarize --exclude=/proc /
71M	/
root@c228adecd486:/# exit

~ >>>  podman run --rm -it debian:10-slim
root@a0adb924e388:/# du --human-readable --summarize --exclude=/proc /
79M	/
root@a0adb924e388:/# apt-get update -qq 2>/dev/null
root@a0adb924e388:/# apt-get install --yes --no-install-recommends libgl1-mesa-glx >/dev/null 2>&1
root@a0adb924e388:/# rm -rf /var/lib/apt/lists/*
root@a0adb924e388:/# du --human-readable --summarize --exclude=/proc /
166M	/
root@a0adb924e388:/# sed -i '/^Depends:/ s/, libgl1-mesa-dri\>//g' /var/lib/dpkg/status
root@a0adb924e388:/# apt-get autoremove -qq --yes
(Reading database ... 6982 files and directories currently installed.)
Removing libgl1-mesa-dri:amd64 (18.3.6-2+deb10u1) ...
Removing libdrm-amdgpu1:amd64 (2.4.97-1) ...
Removing libdrm-intel1:amd64 (2.4.97-1) ...
Removing libdrm-nouveau2:amd64 (2.4.97-1) ...
Removing libdrm-radeon1:amd64 (2.4.97-1) ...
Removing libllvm7:amd64 (1:7.0.1-8+deb10u2) ...
Removing libedit2:amd64 (3.1-20181209-1) ...
Removing libelf1:amd64 (0.176-1.1) ...
Removing libpciaccess0:amd64 (0.14-1) ...
Removing libsensors5:amd64 (1:3.5.0-3) ...
Removing libsensors-config (1:3.5.0-3) ...
Processing triggers for libc-bin (2.28-10) ...
root@a0adb924e388:/# du --human-readable --summarize --exclude=/proc /
86M	/
root@a0adb924e388:/# exit
```

=> If we force it to ignore the dependency on `libgl1-mesa-dri`, we get in the same realm of 6-7 M added.
But, well I have no idea if there are any notable things that would break by this.